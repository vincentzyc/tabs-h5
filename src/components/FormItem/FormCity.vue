<template>
  <div class="flex align-middle form-item">
    <div class="form-label">收货城市</div>
    <div @click="openCityPicker()" class="form-input flex-auto flex align-middle disabled-input">
      <div class="wg-input" :class="{ placeholder: !showValue }">{{ showValue ? showValue : "请选择收货城市" }}</div>
    </div>
    <CityPicker @selected="closePicker" :locationCity="locationCity" ref="domCityPicker" v-model:show="showPicker" />
  </div>
</template>

<script lang="ts">
import { computed, defineComponent, ref } from "vue";
import { checkCity } from "@/composition/business/use-verify-data";

export default defineComponent({
  name: "FormCity",
  props: {
    modelValue: {
      required: true,
      type: Array,
    },
    locationCity: {
      required: true,
      type: Array,
    },
  },
  emits: ["update:modelValue"],
  setup(props, { emit }) {
    const domCityPicker = ref();
    const showPicker = ref(false);
    const showValue = computed(() => {
      return props.modelValue.length > 0 ? props.modelValue.join(" ") : "";
    });

    const openCityPicker = () => {
      showPicker.value = true;
    };
    const closePicker = (val: string[]) => {
      if (Array.isArray(val) && val.length === 3) {
        emit("update:modelValue", val);
        checkCity(val);
      }
    };
    return {
      showPicker,
      showValue,
      domCityPicker,
      checkCity,
      openCityPicker,
      closePicker,
    };
  },
});
</script>
