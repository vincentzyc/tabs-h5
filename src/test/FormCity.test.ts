import { mount } from '@vue/test-utils'
import FormCity from '@/components/FormItem/FormCity.vue'

test('mount FormCity component', async () => {
  expect(FormCity).toBeTruthy()

  const wrapper = mount(FormCity, {
    props: {
      modelValue: '',
      'onUpdate:modelValue': (e: string[]) => wrapper.setProps({ modelValue: e })
    }
  })

  expect(wrapper.html()).toMatchSnapshot()

  // await wrapper.get('input').setValue('测试姓名')

  // expect(wrapper.props('modelValue')).toBe('测试姓名')

  // const emitEvent = wrapper.emitted('update:modelValue')

  // expect(emitEvent).toHaveLength(1)
})