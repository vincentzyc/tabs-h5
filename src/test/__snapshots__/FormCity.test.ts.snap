// Vitest Snapshot v1

exports[`mount FormCity component 1`] = `
"<div class=\\"flex align-middle form-item\\">
  <div class=\\"form-label\\">收货城市</div>
  <div class=\\"form-input flex-auto flex align-middle disabled-input\\">
    <div class=\\"wg-input placeholder\\">请选择收货城市</div>
  </div>
  <!--teleport start-->
  <!--teleport end-->
</div>"
`;
